TowerForge/TowerForge/GameModule/Events/PlayerEvents/WaveSpawnEvent.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  RequestSpawnEvent.swift
+//  TowerForge
+//
+//  Created by Vanessa Mae on 23/3/24.
+//
+
+import Foundation
+
+struct WaveSpawnEvent: TFEvent {
+    let timestamp: TimeInterval
+    let position: CGPoint
+    let entityType: (PlayerSpawnable & TFEntity).Type
+    let player: Player
+
+    /// TODO: might need to change this to something other than throwing fatal error,
+    /// because it could cause accidental application crash, for example iterating through
+    /// a collection of TFEvents.
+    var entityId: UUID {
```

Comment:
Wpuld it be better to modify TFEvent to not require a entityid? Like rename the existing TFEvent to TFEntityEvent and create a new base TFEvent.